aws/data_source_aws_instances_test.go

```diff
@@ -58,108 +62,73 @@ func TestAccAWSInstancesDataSource_instance_state_names(t *testing.T) {
 	})
 }
 
-const testAccInstancesDataSourceConfig_ids = `
-data "aws_ami" "ubuntu" {
-  most_recent = true
-
-  filter {
-    name   = "name"
-    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
-  }
-
-  filter {
-    name   = "virtualization-type"
-    values = ["hvm"]
-  }
-
-  owners = ["099720109477"] # Canonical
-}
-
+func testAccInstancesDataSourceConfig_ids(rName string) string {
+	return composeConfig(
+		testAccLatestAmazonLinuxHvmEbsAmiConfig(),
+		testAccAvailableEc2InstanceTypeForRegion("t3.micro", "t2.micro"),
+		fmt.Sprintf(`
 resource "aws_instance" "test" {
   count         = 3
-  ami           = data.aws_ami.ubuntu.id
-  instance_type = "t2.micro"
+  ami           = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
+  instance_type = data.aws_ec2_instance_type_offering.available.instance_type
 
   tags = {
-    Name = "TfAccTest"
+    Name = %q
   }
 }
 
 data "aws_instances" "test" {
   filter {
     name = "instance-id"
     values = [
-      aws_instance.test.*.id[0],
-      aws_instance.test.*.id[1],
-      aws_instance.test.*.id[2],
+      aws_instance.test[0].id,
+      aws_instance.test[1].id,
+      aws_instance.test[2].id,
```

Comment:
Nit: tal vez a splat could be used here instead? 
```
values = aws_instance.test[*].id

Reply:
Genius!